tests/outlineCompiler_test.py

```diff
@@ -761,6 +761,30 @@ def test_compile_strange_glyph_order(self, quadufo):
         compiler.compile()
         assert compiler.otf.getGlyphOrder() == EXPECTED_ORDER
 
+    def test_compile_reorder_space_glyph(self, FontClass):
+        """
+        Test that ufo2ft always puts .notdef first, and put space second if no
+        explicit glyph order is set.
+        """
+        # This UFO does not have a "public.glyphOrder"
+        ufo = FontClass(getpath("TestSpaceOrder.ufo"))
+
+        EXPECTED_ORDER = [".notdef", "space", "a", "b", "c", "d", "e", "f", "g"]
+        # We can’t assert a specific glyph order here because ufolib2 and
+        # defcon do things differently, so only assert it is not the expected
+        # final re-ordered one.
+        assert list(ufo.keys()) != EXPECTED_ORDER
```

Comment:
if there is a slight chance that this assertion may fail because keys() happened to be ordered the same as the expected order, I would suggest to simply remove this line

Reply:
I haven't tested this, but as far as I can tell by looking at the respective codebases, both ufoLib2 and defcon return an empty list if glyphOrder is not set in the lib, is that not what you found?

Reply:
OK, I think I was tricked by my earlier attempt to create the test font on the fly, that is where the difference happens, not when loading fonts from files.

Reply:
I think I found my source of confusion. We are getting an empty glyphOrder, but code in `makeOfficialGlyphOrder` specifically checks for None, so and empty glyphOrder will prevent reordering.

Reply:
yeah. as I said in the other comment, I think it's better to treat the two (absence vs empty vs not listing 'space' among ordered glyphs) in the same way as a way of saying I don't care where you put my space